### ISSUE TITLE
Time zooming should anchor around the current moment

### ISSUE THREAD

**Issue by gaearon** [forensic-architecture/timemap#231]
Environment
-----------

* Your version (in package.json) or git commit hash: 2cea0fd69e5ba49aabd7d0fb8b8a287a7148c692
* Your operating system and version: macOS

Steps to reproduce (for bugs only)
-----------------------------

1. Switch between different "20 years", "2 years", "3 months", etc

Current Behavior
----------------

The "time zoom" seems to be anchored around the middle of the time bar. E.g. if the middle of the timebar is November 2015, the "zoomed in" version will also use November 2015.

https://user-images.githubusercontent.com/810438/161392651-25d2d6d2-d8ce-4c6a-96f8-84ded545b53a.mov

I believe that this is confusing because switching between two time zoom levels often causes you to completely "lose track" of the currently visible (or even selected) events.

Expected Behavior
-----------------

I would suggest that the horizontal red line which represents the "selected moment" is used as the anchor when switching time zoom levels. In other words, when you switch the time zoom levels, I expect the red line to stay in the same position. This is analogous to how time zooming typically works in e.g. audio editors like Ableton.

----

Disclaimer: I'm not using the tool. It's just something I noticed that I might have interest in fixing.

**Comment by breezykermo** [comment]
Yes, I agree that this is a desirable functionality: selected events should stay center-screen when moving between zoom levels on the timeline. I'd welcome a PR along these lines.
